backend/apis/clients/list-clients.api.js

```diff
@@ -1,23 +1,42 @@
-const { app, databaseError, pool, invalidRequiest } = require("../../server");
+const { app, databaseError, pool, invalidRequest } = require("../../server");
 const mysql = require("mysql");
 
 app.get("/api/clients", (req, res, next) => {
   pool.getConnection((err, connection) => {
     if (err) {
       return databaseError(req, res, err);
     }
-    var queryString = "SELECT ? AS result";
 
-    const query = mysql.format(queryString, [req.query.foo]);
+    var queryString =
+      "SELECT cl.id,cl.firstName,cl.lastName,cl.birthday,ct.zip,ct.primaryPhone,cl.addedBy as addedById,us.firstName as addedByFirstName,us.lastname as addedByLastName,cl.dateAdded FROM clients cl JOIN contactInformation ct ON cl.id = ct.clientId JOIN users us ON cl.addedBy = us.Id WHERE (cl.firstName LIKE ? AND cl.lastName LIKE ?) OR ct.zip = ? ORDER BY cl.firstName,cl.lastName DESC LIMIT 100";
+    if (req.query.page) {
```

Comment:
We should verify that `req.query.page` is a valid number. See [example](https://github.com/JustUtahCoders/comunidades-unidas-internal/blob/d61a30598c0ed44eb7b120ecbfd8938e5aee46c5/backend/apis/clients/get-client.api.js#L15-L19).

We should do this validation check *before getting a database connection*, even though the example I linked to does not. This is because the example I linked to forgot to release the connection before returning.

```js
const validationErrors = checkValid(req.params, nullableValidInteger("page"))
```

The `nullableValidInteger` function is not yet implemented, so you'll have to implement it inside of [validation-utils.js](https://github.com/JustUtahCoders/comunidades-unidas-internal/blob/d61a30598c0ed44eb7b120ecbfd8938e5aee46c5/backend/apis/utils/validation-utils.js)

Reply:
This maybe over my head. check what i did and let me know..

Reply:
Yeah looks good, except for the merge conflict issues I called out in other comments.